harvdev_utils/chado_functions/feature.py

```diff
@@ -0,0 +1,317 @@
+    """Fetch the feature and check the symbol.
+
+    Lookup the feature by uniquename and by symbol and make sure this is the same.
+
+    features are unique wrt:- uniquename, organism_id and type_id
+    So we need to make sure we have all 3 to be safe. (though normally just the uniquename may do)
+
+    uniquename : FBxx0000001 type. Also be aware of things like FBgn0000014:11 which is an exon.
+
+    .. note::
+        If user supplies no type_name or organism, we can get the probable organism form the synonym
+        then lookup just using unique name and report problem if more than one found.
+
+    Args:
+        session (sqlalchemy.orm.session.Session object): db connection  to use.
+
+        uniquename (str): feature uniquename.
+
+        synonym (str): symbol to look up.
+
+        type_name (str) : <optional> cvterm name for the type of feature.
+
+        organism_id (int): <optional> chado organism_id.
+    Returns:
+        Feature object.
+
+    Raises:
+        DataError: if feature cannot be found uniquely.
+    """
+    try:
+        feature = get_feature_by_uniquename(session, uniquename, type_name=type_name, organism_id=organism_id)
+    except NoResultFound:
+        message = "Unable to find Feature with uniquename {}.".format(uniquename)
+        raise DataError(message)
+    except MultipleResultsFound:
+        message = "Found more than feature with this 'uniquename' {}.".format(uniquename)
+        raise DataError(message)
+
+    try:
+        feat_check = feature_symbol_lookup(session, type_name, synonym)
+    except NoResultFound:
+        message = "Unable to find Feature with symbol {}.".format(synonym)
+        raise DataError(message)
+    except MultipleResultsFound:
+        message = "Found more than feature with this symbol {}.".format(synonym)
+        raise DataError(message)
+
+    if feat_check.feature_id != feature.feature_id:
+        message = "Symbol {} does not match that for {}.".format(synonym, uniquename)
+        raise DataError(message)
+    return feature
+
+
+def feature_name_lookup(session, name, organism_id=None, type_name=None, type_id=None):
+    """Get feature by its name.
+
+    Lookup feature using the feature name.
+
+    If no organism_id is given then it will default to Dmel
+
+    Args:
+        session (sqlalchemy.orm.session.Session object): db connection  to use.
+
+        name (str): feature name.
+
+        type_name (str) : <optional> cvterm name for the type of feature.
+
+        organism_id (int): <optional> chado organism_id.
+
+    Returns:
+        Feature object.
+
+    Raises:
+        DataError: if feature not found uniquely.
+    """
+    if type_name and type_id:
+        raise CodingError("Cannot specify type_name and type_id")
+
+    # Default to Dros if no organism specified.
+    if not organism_id:
+        organism_id = get_default_organism_id(session)
+
+    if type_name:
```

Comment:
And here?

Reply:
Also, should they be cached? Might be nice for multiple lookups?

Reply:
Sorry about all this extra work! :P

Reply:
here ? organism ? 

Good point on caching though, I will add that. Things will be easier when Gil does this ;-)

Reply:
Ah spotted the other case.  will fix too

Reply:
This is why we have reviews. So thanks.